src/app/treasure-hunt/operation-costs/operation-costs.component.ts

```diff
@@ -37,12 +53,28 @@ export class OperationCostsComponent implements OnInit {
       this.saveSettings();
     }
     this.treasureHuntSub.unsubscribe();
+    this.zipCodeModalSub.unsubscribe();
+  }
+
+  ngAfterViewInit() {
+    this.zipCodeModalSub = this.zipCodeModal.onShown.subscribe(() => {
```

Comment:
This should be in or called from in ngOnInit()

Reply:
Everywhere else in the app when a modal subscription is called its called in ngAfterViewInit()